### DISCUSSION TITLE
How can I restrict access to files stored in a public bucket?

### DISCUSSION THREAD

**Discussion by mrc-bsllt** [supabase/supabase#20270] [Questions]
Hello everyone!
I have created a public bucket in which I create a folder for each user (renamed with his id), and inside it the avatar.
I put in RLSs for all types of actions, and they work, except for the one for access, which although I restricted only to users whose parent folder must match their id, I can read everyone's.
I attach an image of the RLS in question.

![Screenshot 2024-01-09 alle 14 43 17](https://github.com/supabase/supabase/assets/72490444/f5a933b0-fff1-43af-9797-80850180530c)

**Comment by GaryAustin1** [comment]
You have to use a private bucket for that.  Public bucket means anyone can read the files.

**Reply by mrc-bsllt** [reply]
Hello @GaryAustin1  and thank you for your reply.
As written in the documentation I am using a GET request to access the private bucket https://supabase.com/docs/guides/storage/serving/downloads#private-buckets
But the API returns an error 400 saying "jwt malformed", this is my snippet.

```javascript
const { data: userAvatar } = await useAsyncData("user-avatars", () => (
  $fetch(`${config.public.SUPABASE_URL}/storage/v1/object/authenticated/avatars/${user.value?.id}/${username}-avatar`, {
    headers: {
      authorization: `Bearer ${config.public.SUPABASE_JWT}`,
    }
  })
));
```

SUPABASE_JWT is the JWT secret that I find in the settings of my dashboard.

**Reply by GaryAustin1** [reply]
The jwt needs to be the anon key, service_role key or a user token(jwt).   You say jwt secret which is used to secure the jwts and not a key for use in the API.
![image](https://github.com/supabase/supabase/assets/54564956/289ad3e9-23e9-4035-a5e9-de287e48a9ae)

**Comment by m-tse** [comment]
Hey @mrc-bsllt I want to do something similar.

What exactly is `storage.foldername(name)`? I also want to use a bucket, and have certain items be accessible only by the user.
